module/zfs/spa_stats.c

```diff
@@ -153,9 +152,7 @@ spa_read_history_init(spa_t *spa)
 	ssh->size = 0;
 	ssh->private = NULL;
 
-	(void) snprintf(name, KSTAT_STRLEN, "zfs/%s", spa_name(spa));
-
-	ksp = kstat_create(name, 0, "reads", "misc",
+	ksp = kstat_create("", 0, "reads", "misc",
```

Comment:
From `kstat_create(9F)`:

> kstat_create() is used in conjunction with kstat_install(9F) **to allocate and initialize a kstat(9S) structure**.

>  kstat_create() allocates and performs necessary system initialization of a kstat(9S) structure. kstat_create() **allocates memory for the entire kstat (header plus data), initializes all header fields, initializes the data section to all zeroes**, assigns a unique kstat ID (KID), and puts the kstat onto the system's kstat chain. The returned kstat is marked invalid because the provider (caller) has not yet had a chance to initialize the data section.

>  After a successful call to kstat_create() **the driver must perform any necessary initialization of the data section** (such as setting the name fields in a kstat of type KSTAT_TYPE_NAMED). Virtual kstats must have the ks_data field set at this time. The provider may also set the ks_update, ks_private, and ks_lock fields if necessary.

> Once the kstat is completely initialized, kstat_install(9F) is used to make the kstat accessible to the outside world.

I don't think `kstat_t.ks_module` is part of the "data section", so touching it outside of `kstat_create()` is breaking the API: my preference is leaning towards compatibility and correctness rather than stack optimization.